### ISSUE TITLE
GET-EXTDEV-INFO failed when kiwix is not installed.

### ISSUE THREAD

**Issue by jvonau** [iiab/iiab-admin-console#388]
![Screenshot_2021-04-04_01-06-49](https://user-images.githubusercontent.com/5579710/113500265-2d644d80-94e2-11eb-8e6a-a624c18c1404.png)

>ubuntu@ubuntu:\~$ sudo /opt/admin/cmdsrv/scripts/get_ext_devs.sh
/dev/sda1         258095     55171    202924  22% /media/usb0
/dev/sda2      122651844 109579996  10563420  92% /media/usb1
ubuntu@ubuntu:~$ 

External device is a usb2SDcard adapter containing a pi image.

**Comment by tim-moody** [comment]
what is the iiab host?

**Comment by tim-moody** [comment]
Wrote latest raspios to sdcard in sdcard to usb adapter
On rpi 4 install of a minimal iiab:
booted iiab
stopped iiab-cmdsrv as service and ran locally for debugging
inserted sdcard adapter

```
root@box:~# iiab-cmdsrv-ctl GET-EXTDEV-INFO
message to send:  GET-EXTDEV-INFO
Connecting to server...
Received reply:  {"/media/usb1": {"device": "/dev/sda2", "dev_size_k": "1499232", "dev_used_k": "1118644", "dev_sp_avail_k": "286380", "oer2go_modules": [], "zim_modules": {}}}
```

Started Admin Console. No complaints
removed sdcard
rebooted iiab
inserted sdcard and waited for df to show it
Refreshed Admin Console. No complaints

**Comment by tim-moody** [comment]
So the question is how does my sdcard or env differ from yours.

**Comment by tim-moody** [comment]
>what is the iiab host?

seems to be ubuntu (which?). is rpi 4?
mine is on raspios and my iiab is recent but perhaps not the latest

**Comment by tim-moody** [comment]
did a pull of iiab and reinstall. still no error

**Comment by jvonau** [comment]
Started out life as ubuntu-20.10-lite updated to 21.04 pre-release before starting the iiab install. It's the selection of roles installed, I omitted kiwix from being installed. Captured using the above test method 

>ubuntu@ubuntu:\~$ sudo iiab-cmdsrv-ctl GET-EXTDEV-INFO
message to send:  GET-EXTDEV-INFO
Connecting to server...
No Response from IIAB-CMDSRV in 60.06329584121704 seconds
Traceback (most recent call last):
  File "/usr/bin/iiab-cmdsrv-ctl", line 62, in <module>
    resp = raw_input('Hit Enter to wait another ' + str(REQUEST_TIMEOUT/1000) + ' seconds (anything else to stop): ')
NameError: name 'raw_input' is not defined


>sending message server received from client to worker GET-EXTDEV-INFO id b'\x00k\x8bEk'
sending data message server received from client to worker GET-EXTDEV-INFO id b'\x00k\x8bEk'
sending message server received from client to worker GET-EXTDEV-INFO id b'\x00k\x8bEk'
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.9/threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/admin/cmdsrv/./iiab-cmdsrv3.py", line 686, in cmd_proc_thread
    cmd_resp = cmd_handler(cmd_msg)
  File "/opt/admin/cmdsrv/./iiab-cmdsrv3.py", line 813, in cmd_handler
    resp = avail_cmds[cmd]["funct"](cmd_info)
  File "/opt/admin/cmdsrv/./iiab-cmdsrv3.py", line 1101, in get_external_device_info
    if iiab_ini['kiwix']: # if kiwix installed catalog zims
KeyError: 'kiwix'

Looks to be fixable with some try/exception handling at line 1101.

**Comment by tim-moody** [comment]
unfortunately this error will only be the tip of the iceberg if kiwix is not installed as it is assumed in many places. I guess I need to check for it at install time as I check for nginx.

**Comment by tim-moody** [comment]
closed pending further discussion

**Comment by jvonau** [comment]
was corrected with #390
